7주차: PDF 다운로드 기능, api 연결, 리펙토링

Moderator11 opened the pull request:
# 7주차: PDF 다운로드 기능, api 연결, 리펙토링

## 설명

안녕하세요 멘토님! 이번 주차 작업물 PR 요청 드립니다! develop 브랜치에서 피드백 반영이 늦어질 것 같아 develop-temp로 작업물들을 수합하여 PR을 생성하였습니다.

- **상세 설명**

박수민
- react-pdf/render 라이브러리를 사용하여 PDF 내보내기 기능중 '예제 PDF 렌더링 및 다운로드' 기능을 구현하였습니다.

김성진

- 로그인 기능을 useLoginMutation.ts 를 통해 리액트 쿼리 이용한 방식으로 리팩터링했습니다.
- 회원가입 기능을 useRegisterMutation.ts 를 통해 리액트 쿼리 이용한 방식으로 리팩터링했습니다.
- API 에러 핸들링 커스텀훅을 제작했었으나 리액트 쿼리 적용 과정에서 각 API 호출부에 직접 에러핸들링 로직을 정의하게 되어 삭제했습니다.
- 이후 보호라우팅 로직 적용 예정입니다.
- 이후 다른 API요청(/me 등)에 대해서도 리액트 쿼리 적용시킬 계획입니다.

남규리

- Space 페이지 기능 구현

여행 목록 조회 (GET /v1/plans)
 - useQuery 훅을 사용하여 사용자의 전체 여행 목록을 불러오는 기능을 구현했습니다.
 - API 응답 형식({ content: [...] })에 맞춰 데이터를 파싱하고, 로딩 및 에러 상태를 처리하여 사용자 경험을 개선했습니다.
새 여행 계획 생성 (POST /v1/plans)
 - NewPlanWindow 모달에서 입력받은 정보로 새로운 여행 계획을 생성하는 API 연동 기능을 구현했습니다.
 - 계획 생성 성공 시 queryClient.invalidateQueries를 호출하여 여행 목록이 자동으로 갱신되도록 했습니다.
 
여행 계획 삭제 (DELETE /v1/plans/{id})
 - PlanCard 컴포넌트 내부에서 DeletionConfirmWindow 모달을 통해 삭제를 처리하도록 구조를 개선했습니다.
 - 삭제 성공 시 여행 목록이 자동으로 갱신됩니다.

- Plan 페이지 기능 구현

여행 계획 상세 조회 (GET /v1/plans/{id})
 - useParams로 URL의 planId를 받아와 특정 여행 계획의 상세 정보를 불러오는 기능을 구현했습니다.
 - Router.tsx의 경로를 /plan/:id로 수정하여 동적 라우팅이 가능하도록 했습니다.

- **기타 정보**:

## 관련 이슈(없다면 지워주세요)

<!--- 바로 아래에 이슈 링크를 추가해 주세요.-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Simplified PDF export: single-step modal with direct PDF download, QR code in exports, multiple styled templates, and updated button label ("PDF로 내보내기").
  * Route guards: login/register shown only to guests; plan/space require authentication.

* **Improvements**
  * Profile and plans now load from server; create/edit/delete flows show progress and feedback.
  * Login and registration use mutation flows with clearer toast notifications and better error handling.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SeonJun-Hwang committed:
Merge pull request #35 from kakao-tech-campus-3rd-step3/develop

3 주차: 테크스펙중 누락된 기능 및 UI 구현, 빌드 오류 수정

whyyhyh committed:
Merge branch 'main' of https://github.com/kakao-tech-campus-3rd-step3/Team8_FE into feature/Api-connection

whyyhyh committed:
feat

whyyhyh committed:
feat

whyyhyh committed:
Merge branch 'develop' into feature/api-connection

whyyhyh committed:
feat: api 연결- 임시

whyyhyh committed:
feat: api 연결- 임시

whyyhyh committed:
Merge remote-tracking branch 'origin/develop' into feature/api-connection

whyyhyh committed:
feat: 최신내용반영

whyyhyh committed:
feat: 최신내용반영

whyyhyh committed:
feat: api connection 기반

whyyhyh committed:
feat: api connection 기반

whyyhyh committed:
feat: 서버주소변경

whyyhyh committed:
feat: 서버주소변경

whyyhyh committed:
fix: api connection

whyyhyh committed:
fix

seongjin0320 committed:
fix: 서버 엔드포인트 수정, 프록시 설정 추가

- 서버 로그인/회원가입/리프레시 엔드포인트가 auth/... 으로 바뀌어 명세에 맡게 코드 수정했습니다
- 서버 CORS 이슈 재발로 vite프록시 설정을 위해 vite.config.ts 수정했습니다. CORS이슈 해결시 proxy 설정 부분을 삭제할 수 있습니다

Moderator11 committed:
feat: 5개 pdf 템플릿 형식 추가

Moderator11 committed:
chore: add react-pdf/render

Moderator11 committed:
feat: pdf 링크 컴포넌트 및 테스트용 템플릿 컴포넌트 추가

seongjin0320 committed:
feat: Api 에러핸들링을위한 커스텀훅 추가

- useApiError.ts를 통해 api에러를 표준화 해서 반환합니다
- useToastApiError.ts를 통해 표준화된 에러메시지를 출력합니다.

whyyhyh committed:
Merge branch 'feature/api-connection' of https://github.com/kakao-tech-campus-3rd-step3/Team8_FE into feature/api-connection

whyyhyh committed:
feat: api 호출 수정

whyyhyh committed:
Merge branch 'develop' into feature/api-connection

seongjin0320 committed:
refact: 로그인/회원가입 api 요청 react-query이용방식으로 리팩터링

- 로그인/회원가입 api요청을 react-query를 이용한 방식으로 리팩터링했습니다(useMutation).
- HTTP상태코드를 통해 에러핸들링 로직을 분기했습니다.
- 만약 예외적인 api에러가 일어나면 서버 메시지가 있다면 출력, 아닌경우 기본적인 에러메시지를 출력하도록 했습니다.
- 원래 만들어 뒀던 useApiError.ts와 useToastApiError.ts는 react-query를 도입하는과정에서 불필요하다고 판단해 삭제했습니다.

Moderator11 committed:
feat: 프로필 사진 변경

Moderator11 committed:
feat: Pretendard static 폰트 추가

Moderator11 committed:
feat: PDF 예제 렌더링 및 파일 다운로드 구현

Moderator11 committed:
feat: export 모달 다운로드 링크 추가

Moderator11 committed:
chore: 예제 템플릿 폴더 이동

Moderator11 committed:
Merge branch 'develop' into feature/pdf

Moderator11 committed:
chore: qrcode.react 추가

Moderator11 committed:
Merge pull request #62 from kakao-tech-campus-3rd-step3/feature/pdf

[PDF(1/4)] - PDF 생성 기능

seongjin0320 committed:
Merge pull request #68 from kakao-tech-campus-3rd-step3/feature/react-query-refact

로그인/회원가입 리액트 쿼리 리팩터링

whyyhyh committed:
feat: develop 반영

whyyhyh committed:
Merge remote-tracking branch 'origin/develop-temp' into feature/api-connection

whyyhyh committed:
feat: 프로필 api 연동

whyyhyh committed:
feat: 여행 목록 연동

whyyhyh committed:
Merge pull request #60 from kakao-tech-campus-3rd-step3/feature/api-connection

마이페이지 API 연동

SeonJun-Hwang commented on src/api/axiosInstance.ts:
```diff
@@ -1,22 +1,19 @@
+// ✅ baseURL을 실제 서버 주소로 변경
 const axiosInstance: AxiosInstance = axios.create({
-  baseURL: API_BASE_URL,
+  baseURL: 'http://3.133.89.210:8080', 
```
이부분은 저도 동일합니다. 하드코딩하지 말아주세요

SeonJun-Hwang commented on src/pages/plan/PlanPage.tsx:
```diff
@@ -12,7 +12,7 @@ function PlanPage() {
   useEffect(() => {
-    console.log(`백엔드 plan 정보 요청 (${id})`);
+    // console.log(`백엔드 plan 정보 요청 (${id})`);
   }, [id]);
```
로그를 위한 훅은 제거해주세요

SeonJun-Hwang commented on src/pages/register/hooks/useRegisterForm.ts:
```diff
@@ -19,33 +19,49 @@ export const useRegisterForm = () => {
+      },
+      onError: (e) => {
+        const status = axios.isAxiosError(e) ? e.response?.status : undefined;
+        if (status === 400) {
+          toast.error('입력값을 확인해주세요.');
+          return;
+        }
+        if (status === 409) {
+          toast.error('이미 존재하는 계정입니다.');
+          return;
+        }
+        if (status === 429) {
+          toast.error('요청이 많습니다. 잠시 후 다시 시도하세요.');
+          return;
+        }
+        if (typeof status === 'number' && status >= 500) {
+          toast.error('서버 오류가 발생했습니다. 잠시 후 다시 시도하세요.');
+          return;
+        }
```
일부 에러는 공통 처리를 할 수 있을 거 같아요
또한 error 400으로 들어올 수 있는 경우는 더 많을 것 같아요, 단순히 상태를 보고 처리하는 것 보다, 일부는 서버와 협의후 에러코드를 내리는 포맷으로 변경해야 할거 같아요

seongjin0320 replied:
넵 이 부분은 백엔드와 상의하여 필요한 에러코드를 내릴 수 있는 로직을 설계해보겠습니다!

SeonJun-Hwang commented on src/pages/space/components/planspace/DeletionConfirmWindow.tsx:
```diff
@@ -4,11 +4,30 @@ import { styled } from 'styled-components';
-    console.log(`API call for deleting plan ${plan.id}`);
-    closeModal();
+  const queryClient = useQueryClient();
+  const mutation = useMutation({
+    mutationFn: deletePlanApi,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['plans'] });
+      closeModal();
+    },
+    onError: () => {
+      alert('삭제에 실패했습니다.');
+    },
+  });
```
쿼리 자체는 커스텀훅으로 분리해두는게 더 좋아보여요

SeonJun-Hwang commented on src/pages/space/components/profile/InfoEditWindow.tsx:
```diff
@@ -15,16 +15,35 @@ import { FormInputField } from '@/components/FormInputField';
 
 function InfoEditWindow({ closeModal, member }: ModalPropType & { member: MemberType }) {
+  const queryClient = useQueryClient();
+  const mutation = useMutation({
+    mutationFn: updateMemberInfo,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['memberInfo'] });
+      alert('회원정보 수정이 완료되었습니다!');
+      closeModal();
+    },
+    onError: () => {
+      alert('정보 수정에 실패했습니다.');
+    },
+  });
```
해당 mutation도 다른 사람을 위해 하나의 customHook으로 분리해서 사용하면 좋을거 같아요

SeonJun-Hwang commented on vite.config.ts:
```diff
@@ -9,5 +9,19 @@ export default defineConfig({
+    proxy: {
+      // 프록시 설정 추가(cors 해결되면 삭제 예정)
+      '/api': {
+        target: 'http://3.133.89.210:8080',
```
여기도 동일하게 하드코딩은 제거해주세요

seongjin0320 replied:
axiosInstance.ts와 vite.config.ts에 BASE_URL 하드코딩 제거했습니다!

SeonJun-Hwang reviewed (changes requested):
한주간 고생 많으셨습니다.

저번주보다 코멘트 자체는 적지만, 하드코딩처럼 조금 위험한 코드가 많이 보이네요
전체적으로 수정을 하는 과정이 필요해 보입니다.

또한 useMutation을 이번 PR에서 많이 사용해주셨는데, 컴포넌트 내부에 있는 것 보다
동일한 위치에 커스텀 훅으로 분리하는 것이 더 좋습니다.

지금은 훅이 짧지만 나중에 여러 API를 호출하거나 하는 경우 어떤 목적으로 사용하는 것인지 알기 어렵기 때문에, 훅 이름만 보고도 대강 유추 할 수 있도록 의미있는 이름을 가진 커스텀 훅으로 분리하는 것을 추천드립니다.

또한 React는 기본적으로 컴포넌트 명을 Pascal case로 작성하기 때문에, 대문자로 변수 이름을 시작하게 되면, 컴포넌트로 이해되는 경우가 많습니다.
클래스와 컴포넌트를 제외하고는, 대부분 camelCase로 작성해주시고, 상수는 대문자 기반으로 변경해주세요.
`const object = { MEMO: abc}` 이런형태도 변수이기 때문에 `const object = { memo: abc }` 이런 형태로 변경해주시고, 변경이 안되어야 하는 object 라면 `{ ... } as const`도 같이 붙여주시기 바랍니다.

seongjin0320 committed:
refact: axiosInstance.ts BASE_URL 하드코딩 제거, 리프레시 로직 개선

- axiosInstance.ts 내부에 BASE_URL을 하드코딩 해뒀던 문제를 제거했습니다.
- 리프레시요청에 대해 accessToken을 헤더로 붙이지 않도록 로직 수정했습니다.
- 리프레시 로직 수행 시에, 리프레시 로직 자체에 대해서는 재시도를 하지 않도록 하여 무한루프 문제를 해결했습니다.
- endpoints.ts 의 과거 버전을 제거하고, 최신화된 버전만 남겨놨으며, 엔드포인트들의 구분을 더 직관성있게 할 수 있도록 했습니다.

seongjin0320 committed:
chore: 프록시 설정 제거

seongjin0320 committed:
chore: 리프레시 기능 정상작동 확인을 위한 확인용 코드 추가(이후 삭제 예정)

seongjin0320 committed:
chore: 리프레시 기능 테스트를 위한 버튼 SpacePage.tsx에 추가(정상작동되면 삭제 예정)

seongjin0320 committed:
feature: 보호라우팅을 위한 라우트 가드  추가

seongjin0320 committed:
fix: 랜더시 navigate문제 해결을 위해 라우팅패널 버튼 람다 적용

seongjin0320 committed:
feature: 보호라우팅 적용, SpacaPage에 임시 로그아웃버튼 추가

seongjin0320 committed:
Merge pull request #71 from kakao-tech-campus-3rd-step3/refactor/week7

7주차 피드백 적용, 리팩터링

seongjin0320 committed:
feature: space, plan page로 접근시 로그인 유도, 완료 후 원경로 리디렉트

Moderator11 committed:
feat: PDF 링크 QR 삽입 기능

Moderator11 committed:
feat: PDF 다운로드 모달 수정

Moderator11 committed:
Merge branch 'develop' into feature/pdf-qr

Moderator11 committed:
Merge branch 'develop-temp' into feature/pdf-qr

Moderator11 committed:
Merge pull request #66 from kakao-tech-campus-3rd-step3/feature/pdf-qr

[PDF(2/3)] - PDF QR 생성 및 삽입 기능

seongjin0320 committed:
fix: useLoginForm.ts 에서 any타입 제거

seongjin0320 committed:
Merge pull request #75 from kakao-tech-campus-3rd-step3/feature/protect-routing-cherrypick

Feature/protect routing cherrypick

Moderator11 committed:
feat: 계획 페이지 타이틀 및 설명 fetch 로직

Moderator11 committed:
feat: 여행자 정보 fetch 로직

Moderator11 committed:
Merge pull request #76 from kakao-tech-campus-3rd-step3/feature/plan

계획 정보 fetch 로직 구현

SeonJun-Hwang reviewed (approved):
고오오생 하셨습니다 👍 
다음 PR에서 뵙겠습니다
